File: CSharp/2164.cs
using System;
using System.Collections.Generic;

/***************************************************************************************
 *   * 링크 : https://www.acmicpc.net/problem/2164
 ***************************************************************************************/

namespace BaekJoon
{
    class Program
    {
        static void Main(string[] args)
        {
            Queue<int> q = new Queue<int>();

            int testCase = Convert.ToInt32(Console.ReadLine());

            for (int i = 1; i <= testCase; i++)
            {
                q.Enqueue(i);
            }

            while (q.Count() > 1)
            {
                q.Dequeue();

                q.Enqueue(q.Dequeue());
            }

            Console.WriteLine(q.Dequeue());
        }
    }
}

File: CPlusPlus/2164.cpp
#include <bits/stdc++.h>

#define FASTIO                             \
    std::ios_base::sync_with_stdio(false); \
    std::cin.tie(0);                       \
    std::cout.tie(0);
#define ENDL "\n";

/***************************************************************************************
 *   * 링크 : https://www.acmicpc.net/problem/2164
 ***************************************************************************************/

using namespace std;

int main() {
    FASTIO

    queue<int> q;
    int testCase;
    cin >> testCase;

    for (int i = 1; i <= testCase; i++) {
        q.push(i);
    }

    while (q.size() > 1) {
        q.pop();

        int temp = q.front();
        q.pop();
        q.push(temp);
    }

    cout << q.front() << ENDL;

    return 0;
}